trace.go

```diff
@@ -206,27 +195,20 @@ func SetupTraceOTEL(ctx context.Context, optFns ...TraceOption) (tp *trace.Trace
 		)
 	}
 
-	tp = trace.NewTracerProvider(providerOpts...)
-	shutdownFuncs = append(shutdownFuncs, tp.Shutdown)
-
-	// This option usually needs to be set to false, if we need t test the code
-	// in production environment, we need to set this to true using WithDefaultTracer
-	if opt.setDefaultTracer {
-		otel.SetTracerProvider(tp)
-	}
-
 	// Configure Metric Export to send metrics as OTLP
 	mreader, err := autoexport.NewMetricReader(ctx)
 	if err != nil {
-		err = errors.Join(err, shutdown(ctx))
-		return
+		return nil, err
 	}
+
 	mp := metric.NewMeterProvider(
```

Comment:
How do we shutdown meter? Does `tp.Shutdown`, also shutdown the mp as well?

Reply:
good catch